packages/application-shell/src/components/login/login.js

```diff
@@ -118,210 +92,152 @@ export class Login extends React.PureComponent {
-          method="post"
-          className={styles.form}
-        >
-          {this.renderErrorMessage()}
-          <fieldset className={styles.fieldset}>
-            <Spacings.Stack scale="m">
-              <legend className={styles.title}>
-                <Title>
-                  <FormattedMessage {...messages.title} />
-                </Title>
-              </legend>
-
+      <Formik
+        initialValues={{ email: '', password: '' }}
+        validate={validate}
+        onSubmit={(values, formikBag) => {
+          const addHandlers = promise =>
+            promise.then(
+              result => {
+                formikBag.setSubmitting(false);
+                return result;
+              },
+              error => {
+                formikBag.setSubmitting(false);
+                throw error;
+              }
+            );
+          return this.handleSubmit(values, addHandlers);
+        }}
+        render={formikProps => (
+          <LoginBox>
+            <form onSubmit={formikProps.handleSubmit}>
               <Spacings.Stack scale="m">
-                <Spacings.Stack scale="s">
-                  <label className={styles.label} htmlFor="email">
-                    <FormattedMessage {...messages.email} />
-                  </label>
-                  <input
-                    className={styles['input-text']}
-                    type="text"
-                    name="email"
-                    ref={node => {
-                      this.email = node;
-                    }}
-                    onKeyDown={this.handleKeyPress}
-                  />
-                </Spacings.Stack>
-                <Spacings.Stack scale="s">
-                  <label className={styles.label} htmlFor="password">
-                    <FormattedMessage {...messages.password} />
-                  </label>
-                  <input
-                    className={styles['input-text']}
-                    type="password"
-                    name="password"
-                    onKeyDown={this.handleKeyPress}
+                {this.renderErrorMessage(
+                  !formikProps.isSubmitting && this.state.error
+                )}
+                <Text.Subheadline elementType="h3">
+                  <FormattedMessage {...messages.title} />
+                </Text.Subheadline>
+                <TextField
+                  name="email"
+                  title={this.props.intl.formatMessage(messages.email)}
+                  isAutofocussed={true}
+                  isRequired={true}
+                  value={formikProps.values.email}
+                  touched={formikProps.touched.email}
+                  errors={formikProps.errors.email}
+                  onChange={formikProps.handleChange}
+                  onBlur={formikProps.handleBlur}
+                  isDisabled={formikProps.isSubmitting}
+                  renderError={this.renderEmailErrors}
+                />
+                <TextField
```

Comment:
This should be using the `PasswordField`. Right now the password is shown in plain text!

![image](https://user-images.githubusercontent.com/1765075/50216702-cceb8980-0386-11e9-8b86-54a29126ebd2.png)

Reply:
> We now offer PasswordField commercetools/ui-kit#342 (we need to release first though)

I thought the intend was not to have this blocked. Sorry if I was wrong here.

> using the temporary PasswordField you created

Thanks for spotting that looks like it needs fixing.

Reply:
> I thought the intend was not to have this blocked. Sorry if I was wrong here.

Ah no I'm not saying to block this. I posted it to other readers in the loop that we have the field now (but it's unreleased).

Reply:
Oh shoot, I was importing the wrong component. Fixed now 🤦‍♂️